### PR TITLE
Added OS Customization Spec Support

### PR DESCRIPTION
**Pull request by mvvsubbu**

### CODE REVIEW

File: src/Tools/vmOpsTool/IVMWare.java [hunk 1]
```diff
@@ -92,10 +93,11 @@ void createSnapshot(String vmName, String snapshotName, boolean saveVMMemory, bo
 
     /**
      * @param vmName   name of the virtual machine
+     * @param isCustomizationRequired indiactes whether this operation need to wait for customization or not.
      * @param connData vCenter connection information
      * @throws Exception
      */
-    void powerOnVM(String vmName, ConnectionData connData) throws Exception;
+    void powerOnVM(String vmName, boolean isCustomizationRequired, ConnectionData connData) throws Exception;
```
Comment (kmkumaran) [inline src/Tools/vmOpsTool/IVMWare.java]: Is it possible to determine this state from VM properties instead of having to introduce this variable?

Comment (mvvsubbu) [inline src/Tools/vmOpsTool/IVMWare.java]: Good one, I like this :)

File: src/Tools/vmOpsTool/VMWareImpl.java [hunk 2]
```diff
@@ -268,17 +271,50 @@ private ManagedObjectReference getMorByName(ManagedObjectReference rootContainer
...
+        int waitTimeForToolsRunningStatus = Constants.START_STOP_MAX_WAIT_IN_MINUTES * 90 / 100;
+        int waitTimeForGuestOsHeartBeat = Constants.START_STOP_MAX_WAIT_IN_MINUTES * 10 / 100;
+
+        waitOnMorProperties(vmMor, new String[]{GUEST_TOOLS_RUNNING_STATUS}, new String[]{GUEST_TOOLS_RUNNING_STATUS},
+                new Object[][]{new Object[]{VirtualMachineToolsRunningStatus.GUEST_TOOLS_NOT_RUNNING.value()}},
+                waitTimeForToolsRunningStatus);
+        waitOnMorProperties(vmMor, new String[]{GUEST_HEART_BEAT_STATUS}, new String[]{GUEST_HEART_BEAT_STATUS},
+                new Object[][]{new Object[]{ManagedEntityStatus.GRAY}}, waitTimeForGuestOsHeartBeat);
+
+        System.out.println(String.format("Successfully shutdowned the virtual machine [ %s ].", vmName));
```
Comment (kmkumaran) [inline src/Tools/vmOpsTool/VMWareImpl.java]: shutdowned is not the right word. Rephrase it accordingly.

File: src/Tools/vmOpsTool/VMWareImpl.java [hunk 3]
```diff
@@ -83,6 +88,12 @@ public void cloneVMFromTemplate(String templateName, String vmName, String compu
             throw new Exception(
                     String.format("Failed to create virtual machine [ %s ] using template [ %s ].", vmName, templateName));
         }
+
+        ManagedObjectReference vmMor = getMorByName(targetDCMor, vmName, VIRTUAL_MACHINE, false);
+        if (!customizationSpec.isEmpty()) {
+            waitForOSCustomization(vmName, vmMor, Constants.OS_CUSTOMIZATION_MAX_WAIT_IN_MINUTES);
+        }
+        waitForPowerOnOperation(vmName, vmMor);
```
Comment (kmkumaran) [inline src/Tools/vmOpsTool/VMWareImpl.java]: Does Clone VM do Power ON automatically?

Comment (mvvsubbu) [inline src/Tools/vmOpsTool/VMWareImpl.java]: Yes refer to the change in clone spec

**Comment by kmkumaran** [comment]
Looks fine.

**Pull request merged**